### PR TITLE
fix dropdown onClose when clicking outside menu

### PR DESCRIPTION
**Pull request by rcgillet**

### CODE REVIEW

**Comment by aaronmaturen** [comment]
Add a changeset.

File: .changeset/popular-knives-confess.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"react-magma": patch
```
Comment (aaronmaturen) [inline .changeset/popular-knives-confess.md:2]: react-magma-dom

**Review by aaronmaturen — CHANGES_REQUESTED** [review]

File: .changeset/popular-knives-confess.md (hunk 1 above)
Comment (rcgillet) [inline .changeset/popular-knives-confess.md:2]:
```suggestion
"react-magma-dom": patch
```

**Comment by aaronmaturen** [comment]
My guess is that it's not working because the `handleMenuBlur` is just setting the `isOpen` state directly and not using the helper function that calls the `onClose` prop after `setIsOpen`

Not sure why we even want a global event listener when we can just handle it with the blurs...

**Review by aaronmaturen — APPROVED** [review]

**Pull request merged**